app/main.py

```diff
@@ -1,25 +1,65 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: str, owner: str) -> str:
```

Comment:
The `instance` parameter in the `__get__` method should be of type `Any`, and the `owner` parameter should be of type `type`, not `str`. The `__get__` method is called with the instance of the class and the class itself, not strings.